src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -285,21 +285,35 @@ private static unsafe int IndexOfHtmlEncodingChars(string s, int startPos)
 
         #region UrlEncode implementation
         
-        private static void GetEncodedBytes(byte[] originalBytes, int offset, int count, byte[] expandedBytes)
+        private unsafe static void GetEncodedBytes(byte[] originalBytes, int offset, int count, byte[] expandedBytes)
         {
+            Debug.Assert(originalBytes != null);
```

Comment:
I'd really prefer that these changes were explicit and not compiled out in Retail builds.

I'd also just say to use checked arithmetic when doing operations which might overflow (instead of checking after the fact).